public/app/features/alerting/unified/testSetup/plugins.ts

```diff
@@ -0,0 +1,99 @@
+import { HttpResponse, RequestHandler, http } from 'msw';
+
+import { PluginMeta, PluginType } from '@grafana/data';
+import { config } from '@grafana/runtime';
+
+export function setupPlugins(...plugins: PluginMeta[]): { apiHandlers: RequestHandler[] } {
+  const pluginsRegistry = new Map<string, PluginMeta>();
+  plugins.forEach((plugin) => pluginsRegistry.set(plugin.id, plugin));
+
+  pluginsRegistry.forEach((plugin) => {
+    config.apps[plugin.id] = {
+      id: plugin.id,
+      path: plugin.baseUrl,
+      preload: true,
+      version: plugin.info.version,
+      angular: plugin.angular ?? { detected: false, hideDeprecation: false },
+    };
+  });
+
+  return {
+    apiHandlers: plugins.map((plugin) =>
+      http.get(`/api/plugins/${plugin.id}/settings`, () => {
+        return HttpResponse.json<PluginMeta>(plugin);
+      })
+    ),
+  };
+}
+
+export const plugins: Record<string, PluginMeta> = {
+  slo: {
+    id: 'grafana-slo-app',
```

Comment:
I started making a [method to get a mock plugin here](https://github.com/grafana/grafana/pull/86071/files#diff-7a5ee38dd3276bbc72c7142f49552978f4ee462e5449f25285a5fcf009f230c7R731), we could reuse that? I should have made mine take in `SupportedPlugin`

We should either add `grafana-slo-app` to the `SupportedPlugin` enum, or remove that enum and make it all entirely generic?

Reply:
I think the `SupportedPlugin` enum might be obsolete soon. The new extensions framework allows to support any plugin